### PR TITLE
replaces BANG_PROTECT_2 with a component, also kills OMNITONGUE_2 and flags_2 

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
it's a LITTLE bit iffy but this approach can be used by anything and not just snowflake checks on head and ear items.

e: Now also kills omnitongue_2, a write-only flag
e: Now also kills flags_2

### CODE REVIEW

**Comment by vuonojenmustaturska** [comment]
i was thinking we could rename the now redundantly named flags_1 into atom_flags, and maybe prefix all the other flags we have with the type the flag is from (ATOM_OVERLAY_QUEUED, ITEM_NEEDS_PERMIT, OBJ_EMAGGED etc)

File: code/game/atoms.dm [hunk 1]
```diff
@@ -223,7 +222,7 @@
 
 /atom/proc/emp_act(severity)
 	SendSignal(COMSIG_ATOM_EMP_ACT, severity)
-	if(istype(wires) && !(flags_2 & NO_EMP_WIRES_2))
+	if(istype(wires) && !(flags_1 & NO_EMP_WIRES_1))
```
Comment (optimumtact) [inline code/game/atoms.dm]: I thought you were removing this

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 2]
```diff
@@ -327,6 +323,9 @@
 
 
 /mob/living/carbon/soundbang_act(intensity = 1, stun_pwr = 20, damage_pwr = 5, deafen_pwr = 15)
+	var/list/reflist = list(intensity) // Need to wrap this in a list so we can pass a reference
+	SendSignal(COMSIG_CARBON_SOUNDBANG, reflist)
+	intensity = reflist[1]
```
Comment (optimumtact) [inline code/modules/mob/living/carbon/carbon_defense.dm]: not bad actually, better than anything else I can think of like a double signal chain

Comment (optimumtact) [reply code/modules/mob/living/carbon/carbon_defense.dm]: perhaps we can somehow make this pattern part of the API for signals

**Review by optimumtact — APPROVED** [review]

**Comment by Fox-McCloud** [comment]
Seems a bit excessive to make something into a component that is handled purely by being a bitflag. I guess I don't really see the advantages to this? 

When something has universal behavior and requires complex actions that have to be copy-pasted (ie: materials), it makes sense to use a component. Why use it for a simple flag check though?

File: code/game/atoms.dm (hunk 1 above)
Comment (vuonojenmustaturska) [reply code/game/atoms.dm]: I am, just in another PR, this one already has more than enough stuff

**Comment by vuonojenmustaturska** [comment]
It's a little bit excessive, but spending an /atom (or /obj) level flag for flashbangs to communicate "this item should reduce the flashbang intensity by one when worn on the head or ears" is also less-than-ideal, and lends itself poorly to adjusting the dampening strength per-item.

This approach also lets us have status and reagent effects that provide protection against loud sounds, or a magic sword or whatever. Even organs could be refactored to use this.

Flashbangs happen rarely enough that the additional complexity is not a problem, but I think we gain a valuable knob to adjust game balance and make new content with.

File: code/modules/mob/living/carbon/carbon_defense.dm (hunk 2 above)
Comment (vuonojenmustaturska) [reply code/modules/mob/living/carbon/carbon_defense.dm]: I shamelessly cribbed this from ninjanomnom's musings about signals a day or two ago

**Comment by Fox-McCloud** [comment]
I'm not sure all of those can't be handled by just manipulating flags, to be honest.

As for "but it takes up a whole entire flag!" Yeah, that's fair enough, but....isn't the whole problem with flags avoided by doing what was done for `species_traits`. That is to say; they're no longer bitflags; just a list.

**Comment by optimumtact** [comment]
it depends on how often the flag is checked honestly.

**Comment by Fox-McCloud** [comment]
And how big the list is, I suppose.

**Pull request merged**